### PR TITLE
Anabi 151/ca un utilizator anabi vreau sa modific un bun minimal

### PR DESCRIPTION
**Pull request by Deviad**
Hello,
I was wondering if I should write unit tests or functional tests for this.
This would be something I would really like to learn how to do in .NET. 
Let me know. 

Thank you for your help as well Alex.

### CODE REVIEW

**Comment by Deviad** [comment]
I don't have loads of experience back-end side, but as far as I know creating new objects in a function is considered a bit an antipattern also because it makes creating unit tests harder.

So for example in ModifyAssetHandler.cs

```
            var historicalStageDb = new HistoricalStageDb()
            {
                AddedDate = DateTime.Now,
                StageId = message.StageId,
                UserCodeAdd = "pop",
                EstimatedAmount = message.EstimatedAmount,
                EstimatedAmountCurrency = message.EstimatedAmountCurrency
            };
```
This should come in as a dependency and, therefore, I should register a service that allows me to create such an object. What do you think? @Puiu

File: Anabi/Features/Assets/ModifyAssetHandler.cs [hunk 1]
```diff
@@ -22,11 +23,25 @@ public async Task<MinimalAssetViewModel> Handle(ModifyMinimalAssetObj assetObj)
             var message = assetObj.ModifyMinimalAsset;
             var id = assetObj.Id;
             var assetEntity = await context.FindAsync<AssetDb>(id);
+            
+            
+            var historicalStageDb = new HistoricalStageDb()
```
Comment (Puiu) [inline Anabi/Features/Assets/ModifyAssetHandler.cs]: Please update the historical stage that comes from the GET asset minimal details for the asset, do not add a new one (i know i said something else, but i realized it is not correct now)

**Comment by Puiu** [comment]
You can write unit tests, of course. I am working on the integration tests framework, I will let everyone know when it is ready

File: Anabi/Features/Assets/ModifyAssetHandler.cs (hunk 1 above)
Comment (Deviad) [reply Anabi/Features/Assets/ModifyAssetHandler.cs]: Nevermind I found the issue. When I retrieve the assetEntity here, this is retrieved without the HistoricalStages.
````
var assetEntity = await context.FindAsync<AssetDb>(id);
````

File: Anabi.DataAccess.Ef/AnabiContext.cs [hunk 2]
```diff
@@ -83,5 +83,13 @@ public AnabiContext(DbContextOptions<AnabiContext> options) : base(options)
         public DbSet<CrimeTypeDb> CrimeTypes { get; set; }
 
         public DbSet<PrecautionaryMeasureDb> PrecautionaryMeasures { get; set; }
+        
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
```
Comment (Puiu) [inline Anabi.DataAccess.Ef/AnabiContext.cs]: Why is this necessary?

File: Anabi.DataAccess.Ef/DbModels/AssetDb.cs [hunk 3]
```diff
@@ -26,7 +27,8 @@ public class AssetDb : BaseEntity
 
         public decimal? NecessaryVolume { get; set; }
 
-        public virtual ICollection<HistoricalStageDb> HistoricalStages { get; set; }
+        public  List<HistoricalStageDb> HistoricalStages { get; set; } =
```
Comment (Puiu) [inline Anabi.DataAccess.Ef/DbModels/AssetDb.cs]: This change is not necessary, you can use Include with the lambda function

File: Anabi.DataAccess.Ef/AnabiContext.cs (hunk 2 above)
Comment (Deviad) [reply Anabi.DataAccess.Ef/AnabiContext.cs]: I deleted that. It's just what .NET core shows.
